From 369ffb5be23eca520df99732541d0482da271fd3 Mon Sep 17 00:00:00 2001
From: Lars Op den Kamp <lars@opdenkamp.eu>
Date: Sat, 11 Jul 2015 00:53:17 +0200
Subject: [PATCH] fixed: don't generate an invalid env.h when not built with
 the .git dir present. closes #112

Backported from upstream commit 2f32a9debc1f148b5dfcfc463480f1432bb71725.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 support/git-rev.sh | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/support/git-rev.sh b/support/git-rev.sh
index 5dce3d0..4aadae6 100755
--- a/support/git-rev.sh
+++ b/support/git-rev.sh
@@ -1,5 +1,8 @@
 #!/bin/sh
 
 ## cmake doesn't read the variable when it doesn't end with a newline, and I haven't figured out how to have it add a newline directly...
-echo -n "`git --no-pager log --abbrev=7 -n 1 --pretty=format:"%h"`\n"
-
+if [ -d .git ]; then
+  echo "`git --no-pager log --abbrev=7 -n 1 --pretty=format:"%h"`"
+else
+  echo "<unknown>"
+fi
-- 
2.5.0

